Assignment_2-data-emma-luk.js

```diff
@@ -0,0 +1,101 @@
+﻿// Emma Luk
+// Assignments 2 Using the revealing module pattern, build a Date utility module 
+
+var myDate = (function iife() {
+    var displaysModuleDate = null;
+
+    //private function: date instantiation
+    function _initDate() {
+        return new Date();
+    }
+
+    //check here that date is set: if not, initialize date to now
+    function setDate(date) {
+        if (date === undefined && null) {
+            displaysModuleDate = _initDate();
+        }
+        else if (typeof date === 'number' && !isNaN(date)) {
+            displaysModuleDate = new Date(date);
+        }
+        else if (typeof date === 'object') {

```

Comment:
Using typeof to check against objects isn't advised. You will get the same string 'object' regardless of what type of object you're checking against. For example, `typeof [] === 'object' will also evaluate to true, even though an array literal is not a Date object. Instead, you want to use the instanceof operator. Consult the documentation for instanceof on MDN to see how it works.